spec/support/checkpoint_helpers.rb

```diff
@@ -0,0 +1,28 @@
+
+# frozen_string_literal: true
+
+def new_permit(agent, credential, resource, zone: Checkpoint::DB::Permit.default_zone)
+  Checkpoint::DB::Permit.from(agent, credential, resource, zone: zone)
+end
+
+def agent(type: 'user', id: 'userid')
+  actor = double('actor', agent_type: type, id: id)
+  Checkpoint::Agent.new(actor)
+end
+
+def make_role(name)
+  Checkpoint::Credential::Role.new(name)
+end
+
+def make_permission(name)
+  Checkpoint::Credential::Permission.new(name)
+end
+
+def all_resources
+  Checkpoint::Resource.all
+end
+
+def resource(type: 'resource', id: 1)
+  entity = double('entity', resource_type: type, id: id)
+  Checkpoint::Resource.from(entity)
+end
```

Comment:
Like the policy helpers, I think stuff like this will really improve the ergonomics. I think I want to see what happens when we make it two or three times, to see what different naming and calling semantics emerge before picking a winner.